src/v/cluster/rm_stm.h

```diff
@@ -177,6 +179,55 @@ class rm_stm final : public persisted_stm {
 
     void testing_only_enable_transactions() { _is_tx_enabled = true; }
 
+    struct expiration_info {
+        duration_type timeout;
+        time_point_type last_update;
+
+        time_point_type deadline() const { return last_update + timeout; }
+    };
+
+    struct transaction_info {
+        enum status_t { ongoing, preparing, prepared, initiating };
+        status_t status;
+
+        model::offset first_offset{-1};
+        std::optional<expiration_info> info{std::nullopt};
+
+        ss::sstring get_status() {
+            switch (status) {
+            case ongoing:
+                return "ongoing";
+            case prepared:
+                return "prepared";
+            case preparing:
+                return "preparing";
+            case initiating:
+                return "initiating";
+            }
+        }
+
+        int64_t get_staleness() {
+            if (!info.has_value()) {
+                return -1;
+            }
+
+            auto now = ss::lowres_clock::now();
+            return (now - info->last_update).count();
+        }
+
+        int64_t get_timeout() {
+            if (!info.has_value()) {
+                return -1;
+            }
+
+            return info->timeout.count();
+        }
+    };
+
+    ss::future<result<
+      absl::btree_map<model::producer_identity, rm_stm::transaction_info>>>
+    get_transactions();
```

Comment:
how about `using transaction_set = absl::btree_map<model::producer_identity, rm_stm::transaction_info>` then

`ss::future<result<transaction_set>> get_transactions()`

then use that same typedef in the impl too.